### PR TITLE
COL-387 Allow whiteboard search and filter - back end

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.ets.berkeley.edu/jira/browse/COL-387 (back end component of https://jira.ets.berkeley.edu/jira/browse/COL-380)

Note the new analytics event 'Search whiteboards', which might require additional integration on the Mixpanel side.

### CODE REVIEW

File: node_modules/col-whiteboards/tests/test-whiteboards.js [hunk 1]
```diff
@@ -450,6 +450,111 @@ describe('Whiteboards', function() {
...
+                    verifySearch(users.regularB.client, course, null, whiteboardsB.concat(whiteboardsShared), function() {
+
+                      // When keyword filters are applied, a regular user should see only their whiteboards with matching titles
+                      verifySearch(users.regularA.client, course, {'keywords': 'Berkeley'}, [privateBerkeleyBoard, sharedBerkeleyBoard], function() {
+                        verifySearch(users.regularB.client, course, {'keywords': 'Davis'}, [privateDavisBoard, sharedDavisBoard], function() {
+                          verifySearch(users.regularA.client, course, {'keywords': 'Davis'}, [sharedDavisBoard], function() {
+                            verifySearch(users.regularB.client, course, {'keywords': 'Berkeley'}, [sharedBerkeleyBoard], function() {
+
+                              // An administrator should be able to see all whiteboards associated with a user
+                              verifySearch(users.instructor.client, course, {'user': users.regularA.me.id}, whiteboardsA.concat(whiteboardsShared), function() {
```
Comment (nicolaasmatthijs) [inline node_modules/col-whiteboards/tests/test-whiteboards.js]: This implies that the search would return the 12 whiteboards that were created first and the new whiteboard, which means there would be 13 results. However, as we're not specifying a total number of results to return, we only expect 10.

This should be updated to verify that the search is returning the correct number of results as well.

Comment (nicolaasmatthijs) [inline node_modules/col-whiteboards/tests/test-whiteboards.js]: I noticed that the asset filter tests have the same issue. Can you implement a fix there as well?

**Comment by nicolaasmatthijs** [comment]
This looks really good. Back to @paulkerschen for the comment I left.

**Comment by paulkerschen** [comment]
Result paging verification added in asset and whiteboard searches; back to @nicolaasmatthijs.

**Comment by nicolaasmatthijs** [comment]
Looks good. Merged.

**Pull request merged**